dask/dataframe/io/parquet/core.py

```diff
@@ -300,6 +300,10 @@ def read_parquet(
                 └── └── 04.parquet
 
         Note that the default behavior of ``aggregate_files`` is False.
+    require_extension: str or tuple(str), default (".parq", ".parquet", ".pq")
+        Required file extension for all parquet data files. Other file
+        extensions will be ignored. Note that ``require_extension=False``
+        will skip the file-extension check altogether.
```

Comment:
Good point @jcrist ! (the current doc string is certainly weak).  How about?:

```
        Required file extension for discovered parquet data files. Other
        file extensions will be ignored. This argument only applies when
        ``paths`` corresponds to a directory path, and that directory does
        not contain a '_metadata' file (or ``ignore_metadata_file=True``).
        Note that ``require_extension=False`` will skip this file-extension
        check altogether.

        The purpose of this argument is to ensure that the engine will ignore
        unsupported metadata files (like Spark's '_SUCCESS' and 'crc' files).
        It may be necessary to change this argument if the data files in your
        parquet dataset do not end in ".parq", ".parquet", or ".pq".
```

Reply:
Where did the kwarg name for this option come from? Would it make more sense (in a different PR) to change it to something like `parquet_file_extensions` (where `None` or `True` or something indicated any extension)?

How about this for the first paragraph (the 2nd paragraph is good as is).

```python
A file extension (or an iterable of extensions) to use when finding parquet files in a directory.
Files that don't match this extension(s) will be ignored. This argument only applies when
``paths`` corresponds to a directory and no ``_metadata`` file is present (or
``ignore_metadata_file=True``). Passing in ``require_extension=False`` will treat all files in
the directory as parquet files.
```

Reply:
>Where did the kwarg name for this option come from? Would it make more sense (in a different PR) to change it to something like parquet_file_extensions (where None or True or something indicated any extension)?

It was  a pretty arbitrary name choice from #8274 - I'd be happy to change it. However, this PR is probably the best place to do that, since we are effectively adding a new top-level kwarg (and could raise a warning if the user passes it in the "old" way as "require_extension").

>How about this for the first paragraph

Sure - That looks good to me.

Reply:
> However, this PR is probably the best place to do that

Mmk, if you're happy to do the work here I think we could find a better name. I'll suggest:

```
parquet_file_extension: str, tuple[str], or None (optional)
    A file extension (or an iterable of extensions) to use when finding parquet files in a directory.
    Files that don't match this extension(s) will be ignored. This argument only applies when
    ``paths`` corresponds to a directory and no ``_metadata`` file is present (or
    ``ignore_metadata_file=True``). Passing in ``parquet_file_extension=None`` will treat all files in
    the directory as parquet files.
    
    <your 2nd paragraph here>
```

The default would still be the same tuple you have already.

Reply:
Seems reasonable. cc @martindurant @MrPowers (who were also involved in the original discussion)

Reply:
Yep, that accurately reflects the previous conversation. I've ever met a .pq for a parquet file or otherwise, but I assume someone has.

Reply:
Okay - Latest commit deprecates `require_extension` in favor of a top-level `parquet_file_extension` argument